### ISSUE TITLE
May I ask which dataset are you training on?

### ISSUE THREAD

**Issue by protossw512** [Joker316701882/Deep-Image-Matting#8]
Thank you for providing pretrained model. I have a question regarding to the trainnig dataset. Did you trained on the dataset from original paper or something else?

I tried to do testing on doll.png and troll.png from http://www.alphamatting.com dataset and got pretty bad results, not sure what happened.

I modified test.py code so that I am directly using trimap provided by the dataset, instead of generating trimap from alpha channel.

The segmentation seems to be heavily based on white area on alpha channel.

Here is my result of troll.png:
![image](https://user-images.githubusercontent.com/18194209/31100929-3582afdc-a780-11e7-90c2-754c7c246053.png)

**Comment by Joker316701882** [comment]
@protossw512 
I trained on company's dataset which can be partially seen in test_data folder, it's relatively simple.  Dataset mentioned in paper is more complex so the model training on it can generalize very well.  Unfortunately, I tried to train on dataset mentioned in paper, but it's really really hard to converge. I'll try to improve the performance on it in future. Thank you for your great comments!

**Comment by protossw512** [comment]
@Joker316701882 Thanks for reply, since I am working on university research project, I think I could get the dataset from author. I will update if I could get a better model.

**Comment by Joker316701882** [comment]
@protossw512 
Looking forward to your result! Let me know if you make it!

**Comment by bhack** [comment]
@protossw512 If you have any progress send us an update.

**Comment by hieudt-pixelz** [comment]
In the paper, the authors mentioned this:
"We find images on simple or plain backgrounds (Fig. 2a), including the 27 training images from [25] and every fifth frame from the videos from [26]"

Do you know how can I get that video from [26]?
I glanced over that paper but don't know which video to use.
"[26] E. Shahrian, B. Price, S. Cohen, and D. Rajan. Temporally coherent and spatially accurate video matting. In Proceedings of Eurographics, 2012. 3"

**Comment by Ru-Xiang** [comment]
@protossw512  if you have had a progress, may you share the trained-model with us ?

**Comment by protossw512** [comment]
Yeah, I actually slightly modified loss function and data loading part, fixed some issue with original code that may lead to not good results, and also added refinement stage, getting some promising results, but I am not sure if I am allowed to release pretrained model, since I am training on Adobe dataset and I was not supposed to share it without permission of author.
I haven't got time to refactor my code and post it on github yet. The model itself is not very hard to train, barely need to tune any parameters.

**Comment by Ru-Xiang** [comment]
Thank you for your reply, it will be very appreciated if you share your codes at your convenience.

**Comment by protossw512** [comment]
Yeah, I think I am going to do that during winter break

**Comment by ChengJunqi** [comment]
@protossw512 Hi, can you tell me how to modify the code in detail except for refinement?

**Comment by Joker316701882** [comment]
@protossw512 
Glad to here that you achieved promising results! Can you email me your email address? If it's possible, I would like to discuss about what refinement you have made to achieve good results. Thank : )

**Comment by protossw512** [comment]
@Joker316701882 @ChengJunqi @Ru-Xiang 
Yeah, sure, please email me: helloxinyao.wang###gmail.com
I think one fatal mistake for your original code is the creation of trimap from alpha, you have to perfom eroding and dilation alpha, but you only applied dilation, that's why the network is basically remember the boundaries of your trimap where pixel values are equal to 255.  

There should be other minor issues, I cant remember them tho. I would like to rewrite your code but I do not have any more time to do so, but I can clean them a little bit and upload to my github

**Comment by ricshaw** [comment]
@protossw512 Did you manage to get a better model?

**Comment by foamliu** [comment]
@protossw512 Could you share the improvement in your github?

**Comment by shliang0603** [comment]
@protossw512 Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks

**Comment by aigc-boys** [comment]
@protossw512，Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is 1575262785@qq.com.

**Comment by tsing90** [comment]
@protossw512 I know you may not be able to share your trained model as you explained authorization problem, but would you please share Joker's pretrained model? the link of his pretrained is missing ...... I'll appreciate much if you could do me this favor, thanks. my email: LC2015@ic.ac.uk

**Comment by dailenson** [comment]
> @protossw512，Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is [1575262785@qq.com](mailto:1575262785@qq.com).

have you got the dataset? thanks!

**Comment by jayhhl** [comment]
@protossw512 
Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is [1575262785@qq.com](mailto:332907213@qq.com).

**Comment by jayhhl** [comment]
Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is [332907213@qq.com](mailto:332907213@qq.com).

**Comment by jayhhl** [comment]
@protossw512 
Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is [332907213@qq.com](mailto:332907213@qq.com).

**Comment by ZhangChaoZhong** [comment]
> Yeah, I think I am going to do that during winter break

Can you send me the Adobe datasets.Thanks!My email is 1014295391@qq.com

**Comment by ofrimasad** [comment]
can you please send me the Adobe dataset. The author is not answering emails.
Thanks in advance. ofrimasad@yahoo.com

**Comment by dudumeng** [comment]
Could you send me the Adobe datasets? Thanks! My email is yeqingbit@qq.com

**Comment by linpengzhang** [comment]
@protossw512
Can you send me the Adobe datasets? Thanks! My email is linpeng.zhang@studenti.unipd.it
I'm a CS Student and I need it for MSc thesis

**Comment by sinfere** [comment]
Could you send me the datasets? I'm doing some related researches. Thanks! 

My email is sinfere@gmail.com

**Comment by flyfish01** [comment]
@protossw512

Could you share the Adobe datasets, thank you! my email is haoran.andy###gmail.com

**Comment by parachute0001** [comment]
@protossw512
Could you send me the datasets? I'm doing some related researches. Thanks!
My email is785003851@qq.com

**Comment by parachute0001** [comment]
@protossw512
Could you send me the datasets? I'm doing some related researches. Thanks!
My email is 785003851@qq.com

**Comment by tengda100** [comment]
已经好久不研究这方面的技术了



------------------ 原始邮件 ------------------
发件人: "parachute0001"<notifications@github.com>; 
发送时间: 2019年10月22日(星期二) 上午9:58
收件人: "Joker316701882/Deep-Image-Matting"<Deep-Image-Matting@noreply.github.com>; 
抄送: "Subscribed"<subscribed@noreply.github.com>; 
主题: Re: [Joker316701882/Deep-Image-Matting] May I ask which dataset areyou training on? (#8)




@protossw512
 Could you send me the datasets? I'm doing some related researches. Thanks!
 My email is 785003851@qq.com

**Comment by confident-OH** [comment]
> Thank you for providing pretrained model. I have a question regarding to the trainnig dataset. Did you trained on the dataset from original paper or something else?
> 
> I tried to do testing on doll.png and troll.png from http://www.alphamatting.com dataset and got pretty bad results, not sure what happened.
> 
> I modified test.py code so that I am directly using trimap provided by the dataset, instead of generating trimap from alpha channel.
> 
> The segmentation seems to be heavily based on white area on alpha channel.
> 
> Here is my result of troll.png:
> ![image](https://user-images.githubusercontent.com/18194209/31100929-3582afdc-a780-11e7-90c2-754c7c246053.png)
> 
> I am new to image matting, not sure where I did wrong, do we need to do some preprocessing?

Could you share the Adobe datasets, thank you very much! My email is confidentoh@foxmail.com

**Comment by confident-OH** [comment]
> @protossw512 Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks

Could you share the Adobe datasets, thank you very much! My email is confidentoh@foxmail.com

**Comment by confident-OH** [comment]
> @protossw512
> Can you send me the Adobe datasets? Thanks! My email is [linpeng.zhang@studenti.unipd.it](mailto:linpeng.zhang@studenti.unipd.it)
> I'm a CS Student and I need it for MSc thesis

Could you share the Adobe datasets, thank you very much! My email is confidentoh@foxmail.com

**Comment by kenji0x02** [comment]
@protossw512 Could you send me the Adobe datasets? Thanks! My email is  kenji0x02###gmail.com

**Comment by imocat** [comment]
@protossw512  could you send me the adobe dataset? thanks! my email is imocat###qq.com

**Comment by phper5** [comment]
> Hello, if you can't have the dataset from Adobe, check on pixabay.com. There is a large amount of quite large pictures with transparency and good alpha. For instance, try this search :
> https://pixabay.com/fr/photos/?min_width=640&min_height=640&colors=transparent&image_type=photo&order=latest
> 
> The only issue is that you have to download the images by hand, but with a good methodology you can obtain 300 foreground image in a couple of hour.
> 
> Combined with the backgrounds described in the DIM publications, you can obtain encouraging results without any change to the network.

i think that is a great idea. as i a beginner ,i do not know the mean " Combined with the backgrounds described in the DIM publications"  could you tell me what is the DIM ?

thanks

**Comment by PytaichukBohdan** [comment]
@protossw512
I would appreciate very much if you could  send me the Adobe datasets :)
It is needed very much for my Masters Degree.
My email is:
bohdan.pytaichuk@gmail.com

**Comment by drnemor** [comment]
@protossw512 Could you send me Adobe dataset? Thank you for your work! My email: drnemor@gmail.com

**Comment by mingqizhang** [comment]
I apply the dataset from paper author, but did not get the respond.
Could you please send me the Adobe datasets, my school mail is 
161060075@hdu.edu.cn

**Comment by jaintarun** [comment]
@protossw512 
Hi can you send me the Adobe dataset also? I emailed bprice@adobe.com but I have received no reply. I am doing a project in Image Matting and it will help a lot if I can get the dataset.
My email address is tjain73@yahoo.com

> @Joker316701882 @ChengJunQi @Ru-Xiang
> Yeah, sure, please email me: helloxinyao.wang###gmail.com
> I think one fatal mistake for your original code is the creation of trimap from alpha, you have to perfom eroding and dilation alpha, but you only applied dilation, that's why the network is basically remember the boundaries of your trimap where pixel values are equal to 255.
> 
> There should be other minor issues, I cant remember them tho. I would like to rewrite your code but I do not have any more time to do so, but I can clean them a little bit and upload to my github

**Comment by SvenGroen** [comment]
Has anyone access to the dataset and is able to share it with me. I am working on my bachelor thesis and desperately looking for a benchmark dataset!
Feel free to send it to me:
sgroen@uni-osnabrueck.de

**Comment by cisspbru** [comment]
Hello, could you please send me the Adobe dataset. The author is not answering emails.
I'm a  spbgpu student and I  needed  it very much my email   6032083@mail.ru

**Comment by alexey-trushkov** [comment]
@protossw512 

Hi, would you mind sharing with me Adobe dataset. I'm doing some related research. My email: alexeitrushkov@gmail.com.

**Comment by Yunyung** [comment]
@protossw512
Hello, could you mind send me the Adobe dataset. I'm not get the respond from author.
I'm a CS student and only for research purpose email. Thank a lot! email: s0554014@gm.ncue.edu.tw

**Comment by luckyboy101** [comment]
Can you send me the Adobe datasets.Thanks!My email is 2783000@qq.com

**Comment by ghost** [comment]
@protossw512
Hi! Could you possible share with me Adobe datasets. My email is pseudomo@yandex.ru
Thanks!

**Comment by luckyboy101** [comment]
Hi, I am not allowed to share this dataset due to the adobe license agreement.
Please contact Brian Price (bprice@adobe.com), and you will need to sign an agreement.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Joker316701882/Deep-Image-Matting"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年10月18日(星期天) 上午6:53
收件人:&nbsp;"Joker316701882/Deep-Image-Matting"<Deep-Image-Matting@noreply.github.com&gt;;
抄送:&nbsp;"loske"<2783000@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [Joker316701882/Deep-Image-Matting] May I ask which dataset are you training on? (#8)





 
@protossw512
 Hi! Could you possible share with me Adobe datasets. My email is pseudomo@yandex.ru
 Thanks!

**Comment by qingyang-jiandao** [comment]
@protossw512
Can you send me the Adobe datasets, I have send a email for author of Deep Image Matting, but I didn't get a reply.Thanks. My email is 695450439@qq.com.

**Comment by zpy-number** [comment]
Hi, would you mind sharing with me Adobe dataset. I'm doing some related research. My email:17302632110@163.com

**Comment by luckyboy101** [comment]
Hi, I am not allowed to share this dataset due to the adobe license agreement.
Please contact Brian Price (bprice@adobe.com), and you will need to sign an agreement.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "zpy-number"<notifications@github.com&gt;; 
发送时间: 2021年3月5日(星期五) 凌晨0:35
收件人: "Joker316701882/Deep-Image-Matting"<Deep-Image-Matting@noreply.github.com&gt;; 
抄送: "loske"<2783000@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [Joker316701882/Deep-Image-Matting] May I ask which dataset are you training on? (#8)





 
Hi, would you mind sharing with me Adobe dataset. I'm doing some related research. My email:17302632110@163.com

**Comment by cnnAndBn** [comment]
hi,@Joker316701882,I have a question about the dataset.
in paper ,the fig 2,b shows the alpha result of fig 2 a by using ps tools right? but fig2 c seems to not correspond well to fig 2 c.
as the following image show, the alpha image is very accurate but the corresponding image in fg folder is not good as highlighted in the blue circle. what is the images in the fg folder? how they are generated?
thks
<img width="334" alt="捕获2" src="https://user-images.githubusercontent.com/69999981/113371521-2847b700-9399-11eb-9dc0-5b9e65ec5ade.PNG">

**Comment by heran** [comment]
@protossw512 Could you share the Adobe datasets, thank you very much! My email is wenqiang#xmdqq.com

**Comment by Gundoganfa** [comment]
Could you share the Adobe datasets, thank you very much! My email is gundoganfa@gmail.com

**Comment by AloneGu** [comment]
@protossw512 Could you send me Adobe Matting dataset? Thank in advance! My email: wg631@nyu.edu

**Comment by luckyboy101** [comment]
Hi, I am not allowed to share this dataset due to the adobe license agreement.
Please contact Brian Price ***@***.***), and you will need to sign an agreement.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jackling ***@***.***&gt;; 
发送时间: 2021年6月18日(星期五) 晚上6:04
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Joker316701882/Deep-Image-Matting] May I ask which dataset are you training on? (#8)





 
@protossw512 Could you send me Adobe Matting dataset? Thank in advance! My email: ***@***.***
 
&mdash;

**Comment by FishHe** [comment]
@protossw512 Could you sent me the Adobe dataset. My email: zhangdengsu@163.com
Thank you very much~!

**Comment by foamliu** [comment]
I cannot.



---- Replied Message ----
| From | ***@***.***> |
| Date | 01/06/2023 11:14 |
| To | ***@***.***> |
| Cc | Yang ***@***.***>***@***.***> |
| Subject | Re: [Joker316701882/Deep-Image-Matting] May I ask which dataset are you training on? (#8) |

@protossw512 Could you sent me the Adobe dataset. My email: ***@***.***
Thank you very much~!
